cpp/include/cpu_operations.h

```diff
@@ -400,6 +401,29 @@ class CpuOperations {
      exit(1);
     }
 }
+  static Vector<T> GetOrthogonal(const Vector<T> &a) {
+    if ( a.mean() == 0 )
+     std::cerr << "Vector can not be all zeros!";
+    Vector<T> b(a.size());
+    b.setOnes();
+    Vector<T> projection;
+    projection = ((a.dot(b))/(pow(a.norm(), 2))) * a;
```

Comment:
a.norm() returns |a|^2, there is no need to do another pow() because this would make at |a|^4